Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.swift

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+@MainActor
+public struct OfflineMapAreasView: View {
+    /// The view model for the map.
+    @State private var mapViewModel: MapViewModel
+    
+    /// The action to dismiss the view.
+    @Environment(\.dismiss) private var dismiss: DismissAction
+    
+    
+    public init(map: Map) {
+        self.mapViewModel = MapViewModel(map: map)
+    }
+    
+    public var body: some View {
+        NavigationStack {
+            Form {
+                Section(header: Text("Preplanned Map Areas").bold()) {
+                    preplannedMapAreas
+                }
+                .textCase(nil)
+            }
+            .task {
+                Task {
+                    await mapViewModel.makePreplannedOfflineMapModels()
+                }
```

Comment:
```suggestion
                await mapViewModel.makePreplannedOfflineMapModels()
```
The standalone task block inherits from its context, thus it has the same actor as the view.

I'll tinker with the actor assignments later, but likely we need all the way main actor, as the data model is main actor 🤥